docs/programs/testing.md

```diff
@@ -0,0 +1,252 @@
+---
+title: "Testing with NodeJS"
+description: "Testing native solana programs written with rust using NodeJS"
+---
+
+When developing programs on Solana, ensuring their correctness and reliability
+is crucial. Until now devs have been using solana-test-validator for testing.
+This document covers testing your Solana program with NodeJS
+using `solana-bankrun`.
+
+## Overview
+
+There are two ways to test programs on Solana:
+
+1. [solana-test-validator](https://docs.solanalabs.com/cli/examples/test-validator):
+   That spins up a local emulator of the Solana Blockchain on your local machine
+   which receives the transactions to be processed by the validator.
+2. BanksClient-based test framework for SBF(Solana Bytecode Format) programs:
+   Bankrun is a framework that simulates a Solana bank’s operations, enabling
+   developers to deploy, interact with, and assess the behaviour of programs
+   under test conditions that mimic the mainnet. It helps set up the test
+   environment and offers tools for detailed transaction insights, enhancing
+   debugging and verification. With the client, we can load programs, and
+   simulate and process transactions seamlessly.
+   [solana-program-test](https://docs.rs/solana-program-test) (Rust),
+   [solana-bankrun](https://github.com/kevinheavey/solana-bankrun) (Rust,
+   JavaScript), [anchor-bankrun](https://www.npmjs.com/package/anchor-bankrun)
+   (Anchor, JavaScript),
+   [solders.bankrun](https://kevinheavey.github.io/solders/api_reference/bankrun.html)
+   (Rust, Python) are examples of the BanksClient-based testing framework.
+
+   ```
+   Note:
+
+   [pnpm create solana-program](https://github.com/solana-program/create-solana-program) can help you generate JS and Rust clients including tests.
+   Anchor is not yet supported.
+   ```
+
+In this guide, we are using Solana Bankrun. `Bankrun` is a superfast, powerful,
+and lightweight framework for testing Solana programs in NodeJS.
+
+- The biggest advantage of using Solana Bankrun is that you don’t have to set up
+  an environment to test programs like you’d have to do while using the
+  solana-test-validator. Instead, you can do that with a piece of code inside
+  the tests.
+- It also dynamically setting time and account data, which isn’t possible with
+  solana-test-validator
+
+## Installation
+
+Add `solana-bankrun` as a dev dependency to your node project. If your Solana
+program is not a node project yet, you can initialize it using `npm init`.
+
+```bash
+npm i -D solana-bankrun
+```
+
+## Usage
+
+### Program Directory
+
+Firstly, the program's `.so` file must be present in one of the following
+directories:
```

Comment:
Should you add a section on how to download `.so`s from mainnet?

Reply:
will be good if we add that in future, since the guide suffices for hello world.